File: app/src/main/java/com/erif/contentloader/MainActivity.java
package com.erif.contentloader;

import android.content.Intent;
import android.os.Bundle;
import android.widget.Button;

import androidx.appcompat.app.AppCompatActivity;

import com.erif.contentloader.example.ActivityCombine;
import com.erif.contentloader.example.ActivityBanner;
import com.erif.contentloader.example.ActivityGrid;
import com.erif.contentloader.example.ActivityHorizontal;
import com.erif.contentloader.example.ActivityShimmer;
import com.erif.contentloader.example.ActivityVertical;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        if (getSupportActionBar() != null) {
            getSupportActionBar().setTitle("Content Loader Animation");
            getSupportActionBar().setSubtitle("https://github.com/eriffanani/ContentLoader");
        }

        onClick(R.id.btnVertical, ActivityVertical.class);
        onClick(R.id.btnShimmer, ActivityShimmer.class);
        onClick(R.id.btnHorizontal, ActivityHorizontal.class);
        onClick(R.id.btnGrid, ActivityGrid.class);
        onClick(R.id.btnBanner, ActivityBanner.class);
        onClick(R.id.btnCombine, ActivityCombine.class);

    }

    private void onClick(int id, Class<?> destination) {
        getButton(id).setOnClickListener(view -> intent(destination));
    }

    private void intent(Class<?> destination) {
        Intent intent = new Intent(this, destination);
        startActivity(intent);
    }

    private Button getButton(int id) {
        return findViewById(id);
    }

}

File: app/src/main/java/com/erif/contentloaderdemo/MainActivity.kt
package com.erif.contentloaderdemo

import android.content.Intent
import android.os.Bundle
import android.widget.Button
import androidx.appcompat.app.AppCompatActivity
import com.erif.contentloaderdemo.example.ActivityBanner
import com.erif.contentloaderdemo.example.ActivityCombine
import com.erif.contentloaderdemo.example.ActivityGrid
import com.erif.contentloaderdemo.example.ActivityHorizontal
import com.erif.contentloaderdemo.example.ActivityShimmer
import com.erif.contentloaderdemo.example.ActivityVertical

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        supportActionBar?.let {
            it.title = "Content Loader Animation"
            it.subtitle = "https://github.com/eriffanani/ContentLoader"
        }
        onClick(R.id.btnVertical, ActivityVertical::class.java)
        onClick(R.id.btnShimmer, ActivityShimmer::class.java)
        onClick(R.id.btnHorizontal, ActivityHorizontal::class.java)
        onClick(R.id.btnGrid, ActivityGrid::class.java)
        onClick(R.id.btnBanner, ActivityBanner::class.java)
        onClick(R.id.btnCombine, ActivityCombine::class.java)
    }

    private fun onClick(id: Int, destination: Class<*>) {
        button(id)?.setOnClickListener {
            intent(destination)
        }
    }

    private fun intent(destination: Class<*>) {
        val intent = Intent(this, destination)
        startActivity(intent)
    }

    private fun button(id: Int): Button? {
        return findViewById(id)
    }

}